pkg/alertmanager/operator.go

```diff
@@ -654,6 +657,23 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 
 	level.Info(logger).Log("msg", "sync alertmanager")
 
+	// Check if operator has the permission to read the storage class.
+	if c.canReadStorageClass {
+		// Check if the AlertManager Specs is using a Volume Claim.
+		if am.Spec.Storage.VolumeClaimTemplate.Spec.StorageClassName != nil {
```

Comment:
see https://kubernetes.io/docs/concepts/storage/persistent-volumes/#class-1, nil and empty storage class names are  a bit different but the operator can't make any conclusion in both cases.

```suggestion
	if c.canReadStorageClass && am.Spec.Storage.VolumeClaimTemplate.Spec.StorageClassName != nil && *am.Spec.Storage.VolumeClaimTemplate.Spec.StorageClassName != "" {
```

Reply:
Thank you for the suggestion and pointing about the empty storage class name, will apply the changes.